centralizedmetrics/metrics.go

```diff
@@ -0,0 +1,138 @@
+package centralizedmetrics
+
+import (
+	"database/sql"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/status-im/status-go/centralizedmetrics/common"
+	"github.com/status-im/status-go/centralizedmetrics/providers"
+)
+
+var defaultPollInterval = 10 * time.Second
+
+type MetricsInfo struct {
+	Enabled       bool `json:"enabled"`
+	UserConfirmed bool `json:"userConfirmed"`
+}
+
+type MetricRepository interface {
+	Poll() ([]common.Metric, error)
+	Delete(metrics []common.Metric) error
+	Add(metric common.Metric) error
+	Info() (*MetricsInfo, error)
+	ToggleEnabled(isEnabled bool) error
+}
+
+type MetricService struct {
+	repository MetricRepository
+	processor  common.MetricProcessor
+	ticker     *time.Ticker
+	done       chan bool
+	started    bool
+	wg         sync.WaitGroup
+}
+
+func NewDefaultMetricService(db *sql.DB) *MetricService {
+	repository := NewSQLiteMetricRepository(db)
+	processor := providers.NewMixpanelMetricProcessor(providers.MixpanelAppID, providers.MixpanelToken, providers.MixpanelBaseURL)
+	return NewMetricService(repository, processor, defaultPollInterval)
+}
+
+func NewMetricService(repository MetricRepository, processor common.MetricProcessor, interval time.Duration) *MetricService {
+	return &MetricService{
+		repository: repository,
+		processor:  processor,
+		ticker:     time.NewTicker(interval),
+		done:       make(chan bool),
+	}
+}
+
+func (s *MetricService) Start() {
+	if s.started {
+		return
+	}
+	s.wg.Add(1)
+	s.started = true
+	go func() {
+		defer s.wg.Done()
+		for {
+			select {
+			case <-s.done:
+				return
+			case <-s.ticker.C:
+				s.processMetrics()
+			}
+		}
+	}()
+}
+
+func (s *MetricService) Stop() {
+	if !s.started {
+		return
+	}
+	s.ticker.Stop()
+	s.done <- true
```

Comment:
closing would be even more specific
```suggestion
	close(s.done)
```

Reply:
we reuse the channel in case of `Start`/`Stop`, and I am always a bit wary of closing (you don't have to close if it's going to be garbage collect afaik) as it might be causing panics, which crashes the app and I don't think the code is  bullet proof in case of multiple/start/stop etc